src/components/JUPTable/JUPTable.tsx

```diff
@@ -0,0 +1,184 @@
+import { TableRowsPerPageOptions } from "utils/common/constants";
+
+// might still want to use this concept
+// function createWidgetRow(date: string, qty: number, toFrom: string): Data {
+//   return {
+//     date,
+//     qty,
+//     toFrom,
+//   };
+// }
+
+interface ITableTitleProps {
+  title: string;
+}
+const TableTitle: React.FC<ITableTitleProps> = ({ title }) => {
+  return (
+    <>
+      {
+        <TitleText variant="h6" id="tableTitle">
+          <SLink href="/transactions">{title}</SLink>
+        </TitleText>
+      }
+    </>
+  );
+};
+
+function descendingComparator<T>(a: T, b: T, orderBy: keyof T) {
+  if (b[orderBy] < a[orderBy]) {
+    return -1;
+  }
+  if (b[orderBy] > a[orderBy]) {
+    return 1;
+  }
+  return 0;
+}
+
+type Order = "asc" | "desc";
+
+function getComparator<Key extends keyof ITransaction>(
+  order: Order,
+  orderBy: Key
+): (a: { [key in Key]: number | string }, b: { [key in Key]: number | string }) => number {
+  return order === "desc" ? (a, b) => descendingComparator(a, b, orderBy) : (a, b) => -descendingComparator(a, b, orderBy);
+}
+
+interface IEnhancedTableProps {
+  onRequestSort: (event: React.MouseEvent<unknown>, property: keyof Data) => void;
+  order: Order;
+  orderBy: string;
+  rowCount: number;
+  headCells: Array<IHeadCellProps>;
+}
+
+const EnhancedTableHead: React.FC<IEnhancedTableProps> = ({ onRequestSort, order, orderBy, rowCount, headCells }) => {
+  const createSortHandler = (property: keyof Data) => (event: React.MouseEvent<unknown>) => {
+    onRequestSort(event, property);
+  };
+
+  return (
+    <TableHead>
+      <TableRow>
+        {headCells.map((headCell) => (
+          <TableCell
+            key={headCell.id}
+            align={headCell.numeric ? "right" : "left"}
+            padding={headCell.disablePadding ? "none" : "normal"}
+            sortDirection={orderBy === headCell.id ? order : false}
+          >
+            <TableSortLabel
+              active={orderBy === headCell.id}
+              direction={orderBy === headCell.id ? order : "asc"}
+              onClick={createSortHandler(headCell.id)}
+            >
+              {headCell.label}
+              {orderBy === headCell.id ? (
+                <Box component="span" sx={visuallyHidden}>
+                  {order === "desc" ? "sorted descending" : "sorted ascending"}
+                </Box>
+              ) : null}
+            </TableSortLabel>
+          </TableCell>
+        ))}
+      </TableRow>
+    </TableHead>
+  );
+};
+
+interface IJUPTableProps {
+  headCells: Array<IHeadCellProps>;
+  rows: Array<React.ReactElement>;
+  children?: Array<React.ReactElement>;
+  title: string;
+}
+
+const JUPTable: React.FC<IJUPTableProps> = ({ children, headCells, rows, title }) => {
+  const [rowsPerPage, setRowsPerPage] = React.useState(3);
+  const [order, setOrder] = React.useState<Order>("asc");
+  const [orderBy, setOrderBy] = React.useState<any>("date");
```

Comment:
i'm assuming date is expected to always exist on each table. mentioning it as i'm unsure how generic you wanted this component.

Reply:
that might remain the case but i'm not sure, this was just an oversight. I'll leave it for now and deal with it when I don't have a "date" to sort by in a future table